pkg/harvester/client/server.go

```diff
@@ -1,33 +1,40 @@
 package client
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
 	"net/http"
 
+	"github.com/HewlettPackard/galadriel/pkg/common"
 	"github.com/HewlettPackard/galadriel/pkg/common/telemetry"
 	"github.com/sirupsen/logrus"
 )
 
+var contentType = "application/json"
+
 // GaladrielServerClient represents a client to connect to Galadriel Server
 type GaladrielServerClient interface {
 	GetUpdates(context.Context) ([]string, error)
-	PushUpdates(context.Context, []string) error
+	PostBundle(context.Context, *common.PostBundleRequest) error
 	Connect(ctx context.Context, token string) error
 }
 
 type client struct {
-	address string
 	c       http.Client
+	address string
+	token   string
 	logger  logrus.FieldLogger
 }
 
-func NewGaladrielServerClient(address string) (GaladrielServerClient, error) {
+func NewGaladrielServerClient(address, token string) (GaladrielServerClient, error) {
 	return &client{
 		c:       *http.DefaultClient,
-		address: address,
+		address: fmt.Sprintf("http://%s", address),
```

Comment:
since we're using the default client, we don't need to have the http url prefix

```suggestion
		address: fmt.Sprintf("%s", address),
```